gammapy/spectrum/spectrum_analysis.py

```diff
@@ -385,44 +534,71 @@ def write_ogip(self, phafile=None, bkgfile=None, rmffile=None, arffile=None,
         outdir.mkdir(exist_ok=True)
 
         if phafile is None:
-            phafile = outdir / "pha_run{}.pha".format(self.obs)
+            phafile = "pha_run{}.pha".format(self.obs)
         if arffile is None:
-            arffile = outdir / "arf_run{}.fits".format(self.obs)
+            arffile = "arf_run{}.fits".format(self.obs)
         if rmffile is None:
-            rmffile = outdir / "rmf_run{}.fits".format(self.obs)
+            rmffile = "rmf_run{}.fits".format(self.obs)
         if bkgfile is None:
-            bkgfile = outdir / "bkg_run{}.fits".format(self.obs)
-
-        if self.pha is not None:
-            self.pha.write(str(phafile), bkg=str(bkgfile), arf=str(arffile),
-                           rmf=str(rmffile), clobber=clobber)
-        if self.bkg is not None:
-            self.bkg.write(str(bkgfile), clobber=clobber)
-        if self.arf is not None:
-            self.arf.write(str(arffile), energy_unit='keV', effarea_unit='cm2',
-                           clobber=clobber)
-        if self.rmf is not None:
-            self.rmf.write(str(rmffile), energy_unit='keV', clobber=clobber)
-
-    def write_all_ogip_data(self, outdir):
-        """Perform all step to provide the OGIP data for a sherpa fit
+            bkgfile = "bkg_run{}.fits".format(self.obs)
+
+        self.on_vector.write(str(outdir/phafile), bkg=str(bkgfile), arf=str(arffile),
+                             rmf=str(rmffile), clobber=clobber)
+        self.off_vector.write(str(outdir/bkgfile), clobber=clobber)
+        self.effective_area.write(str(outdir/arffile), energy_unit='keV',
+                                  effarea_unit='cm2', clobber=clobber)
+        self.energy_dispersion.write(str(outdir/rmffile), energy_unit='keV',
+                                     clobber=clobber)
+
+    def plot_exclusion_mask(self, size=None, **kwargs):
```

Comment:
Pass `ax` so that it's possible to easily create this plot as one of several panels in a larger Figure?